tests/subsys/llext/simple/src/export_dependent_ext.c

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2024 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Test importing symbols, exported by other LLEXTs */
+
+#include <zephyr/ztest.h>
+#include <zephyr/llext/symbol.h>
+#include <zephyr/sys/util.h>
+extern int test_dependency(short a, short b);
+
+void test_dependent(void)
+{
+	unsigned int half_ptr_bits = sizeof(uintptr_t) * 8 / 2;
+	unsigned int mask = BIT(half_ptr_bits) - 1;
+	short a = mask & (uintptr_t)test_dependent;
+	short b = mask & ((uintptr_t)test_dependent >> half_ptr_bits);
```

Comment:
I don't think `short` is correct on 64 bit arches given the values involved (although the assert will be fine). 
IMHO a simpler calculation with constant values would make the logic clearer and still test proper linking and argument passing.

Reply:
@pillo79 yeah, I was thinking about it too - if addresses are 64-bit, then IIRC `int` can still be 32 bits. So I wanted to replace `int` with `long` and `short` with `int`, would that be enough?